test/test_multidevice_sharding.cpp

```diff
@@ -0,0 +1,105 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#ifdef NVFUSER_DISTRIBUTED
+#include <disjoint_set.h>
+#include <fusion.h>
+#include <fusion_segmenter.h>
+#include <gtest/gtest.h>
+#include <multidevice/executor.h>
+#include <multidevice/utils.h>
+#include <ops/all_ops.h>
+#include <test/multidevice.h>
+#include <test/validator.h>
+
+namespace nvfuser {
+
+class ShardingTest : public MultiDeviceTest {};
+
+TEST_F(ShardingTest, UnshardedGlobalInput) {
```

Comment:
nit: please have at least one-line description about what this test is meant to verify and exercise.